graphql-parser/src/lexer.rs

```diff
@@ -2,14 +2,23 @@
 use std::str::CharIndices;
 use std::iter::Peekable;
 
-#[derive(Debug, PartialEq, Eq)]
+#[derive(PartialEq, Debug)]
 pub enum Token<'a> {
-  Eof,
+  Eof(usize, usize),
+  Punctuator(Punctuator, usize),
```

Comment:
Range is exclusive in the upper bound so I would end up having `+ 1` all over the code